### DISCUSSION TITLE
monai.deploy.exceptions.ItemNotExistsError: A predictor of the model is not set. for the custom model i created using monai label.

### DISCUSSION THREAD

**Discussion by ArpanGyawali** [Project-MONAI/monai-deploy-app-sdk#444] [General]
My task: Skull stripping
What I did:

Rewrote the config, infers, and trainers file of the segmentation-spleen model within radiology by changing labels, roi, and transforms as required and renamed it as segmentation-brain.
Then, I annotated my own dataset for skull stripping using 3D slicer. The only label used was "brain"
Then I trained the images and its label using the custom segmentation-brain model using monai label extension in 3d slicer and got a dice score of about 0.98 and I am happy with the result.
Now I am trying to deploy the model for inference using monai-deploy-app-sdk.

For that, I used the model.pt file saved inside the apps/radiology/models/segmentation brain directory in the command
python app.py -i input_dir -o output_dir -m model.pt
But I am getting the error: ItemNotExistsError: A predictor of the model is not set
How can I get the saved inference model file that can be used while deploying? Do we need a torchscript file of the model? If yes then how to get that?

Any help would be very appreciated.

**Comment by MMelQin** [comment]
The MONAI Deploy App SDK model network loader supports TorchScript, so the model needs to be traced/scripted to a TorchScript file. Better yet, if a MONAI Bundle is created (assuming just with pre_processing, post_processing, and infer), the built MONAI Bundle Inference operator can be to automate these steps in the app.

The least preferred way is to write a custom inference operator which does the model loading from the well-known path, and handle the pre and post processing steps.

**Reply by ArpanGyawali** [reply]
Thanks, @MMelQin 
yes, I have written a custom inference operator with pre and post transforms steps.
Wouldnt I need to trace/script the model to be TorchScript file if I create MONAI bundle and used buildin MONAI bundle Inference operator?

I have a model file **model.pt** which is automatically provided to me by monai label after I trained the custom segmentation model using monai_label extension, so i don't know the architecture of the model and how it is accessed in the background by monai label. How can i trace/script this model.pt file to TorchScript and used it for the deployment so that i dont get that error?

**Reply by ArpanGyawali** [reply]
Also, @MMelQin 
is it okay if i create a MONAI bundle with just inference.json from spleen_ct_segmentation bundle as reference?
do i still need to trace/script the model into TorchScript file?

**Reply by MMelQin** [reply]
MONAI Bundle documentation can be found [here](https://docs.monai.io/en/latest/bundle_intro.html), and it is a good read. A MONAI Bundle is a TorchScript with extra metadata to allow the application to automates the preprocessing and such.
